Library/Homebrew/api.rb

```diff
@@ -140,5 +170,31 @@ def self.write_names_file(names, type, regenerate:)
 
       false
     end
+
+    sig { params(json_data: Hash).returns([T::Boolean, T.any(String, Array, Hash)]) }
+    private_class_method def self.verify_and_parse_jws(json_data)
+      signatures = json_data["signatures"]
+      homebrew_signature = signatures&.find { |sig| sig.dig("header", "kid") == "homebrew-1" }
+      return false, "key not found" if homebrew_signature.nil?
+
+      header = JSON.parse(Base64.urlsafe_decode64(homebrew_signature["protected"]))
+      if header["alg"] != "PS512" || header["b64"] != false # NOTE: nil has a meaning of true
+        return false, "invalid algorithm"
+      end
+
+      require "openssl"
+
+      pubkey = OpenSSL::PKey::RSA.new(File.read("#{__dir__}/api/homebrew-1.pem"))
```

Comment:
Thoughts on changing this to a repo-relative path?

```suggestion
      pubkey = OpenSSL::PKey::RSA.new(HOMEBREW_LIBRARY/"Homebrew/api/homebrew-1.pem"))
```

No strong feelings but I find that easier to understand, personally.